build.sh

```diff
@@ -114,7 +114,12 @@ function do_init
 function prepare_build_dir
 {
   TYPE=$1
-  mkdir -p ${TOPDIR}/build_${TYPE} && cd ${TOPDIR}/build_${TYPE}
+  mkdir -p ${TOPDIR}/build_${TYPE}
+  if [ ! -e build ]; then
```

Comment:
I think the build dir should be created every time.
Take a look at this scenario.
1. I build miniob with debug mode and I got build_debug dir and then build.sh link the build to build_debug.
2. I want to build miniob with release mode and I got build_release dir but build.sh would not work as expected.

Reply:
> I think the build dir should be created every time. Take a look at this scenario.
> 
> 1. I build miniob with debug mode and I got build_debug dir and then build.sh link the build to build_debug.
> 2. I want to build miniob with release mode and I got build_release dir but build.sh would not work as expected.

Done. 

I find that there is a same `prepare_build_dir` function at Line 65, what's the difference between them?

Reply:
Oh... It's a mistake.